extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -142,16 +147,27 @@ export class Bind {
    * @private
    */
   initialize_() {
-    this.scanPromise_ = this.scanBody_(this.ampdoc.getBody());
-    this.scanPromise_.then(results => {
-      const {boundElements, bindings, expressionToElements} = results;
+    this.addBindingsForSubtree(this.ampdoc.getBody());
+  }
 
-      this.boundElements_ = boundElements;
+  /**
+   * Scans the substree rooted at `rootElement` and adds bindings for nodes
+   * that contain bindable elements. This function is not idempotent. To remove
+   * bindings for a subtree, see #removeBindingsForSubtree. Returns a promise
+   * that resolves after bindings have been added.
+   * @param {!Element} rootElement
+   * @return {Promise}
+   */
+  addBindingsForSubtree(rootElement) {
```

Comment:
Mark all private methods with `@private` attribute and trailing underscore. This allows Closure compiler to rename the method during minimization and other optimizations like DCE.